lib/spack/llnl/util/filesystem.py

```diff
@@ -1150,16 +1197,11 @@ def traverse_tree(source_root, dest_root, rel_path="", **kwargs):
             ``src`` that do not exit in ``dest``. Default is True
         follow_links (bool): Whether to descend into symlinks in ``src``
     """
-    follow_nonexisting = kwargs.get("follow_nonexisting", True)
-    follow_links = kwargs.get("follow_link", False)
```

Comment:
We document `follow_links` but we get `follow_link` (missing `s`) from the dictionary